packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,493 @@
+      list3 = children.sublist(bestIndex + 1);
+      list3Size = computeSize(list3, 0, list3.length - 1);
+    } else {
+      // Put all data in l2 and none in l3.
+      list2 = children.sublist(pivotIndex + 1);
+      list2Size = computeSize(list2, 0, list2.length - 1);
+
+      final double pdAndList2Ratio = (pSize + list2Size) / totalSize;
+      final double pdRatio = pSize / (pSize + list2Size);
+      pivotBestWidth =
+          isHorizontalRectangle ? pdAndList2Ratio * width : pdRatio * width;
+      pivotBestHeight =
+          isHorizontalRectangle ? pdRatio * height : pdAndList2Ratio * height;
+    }
+
+    // Contruct list 1 sub-treemap.
+    final double list1SizeRatio = list1Size / totalSize;
+    final double list1Width =
+        isHorizontalRectangle ? width * list1SizeRatio : width;
+    final double list1Height =
+        isHorizontalRectangle ? height : height * list1SizeRatio;
+    List<Cell> list1Cells;
+    list1Cells = buildTreemap(
+      list1,
+      list1Width,
+      list1Height,
+      x,
+      y,
+    );
+
+    // Construct list 2 sub-treemap.
+    final double list2Width =
+        isHorizontalRectangle ? pivotBestWidth : width - pivotBestWidth;
+    final double list2Height =
+        isHorizontalRectangle ? height - pivotBestHeight : pivotBestHeight;
+    final double list2XCoord = isHorizontalRectangle ? x + list1Width : x;
+    final double list2YCoord =
+        isHorizontalRectangle ? y + pivotBestHeight : y + list1Height;
+    List<Cell> list2Cells;
+    list2Cells = buildTreemap(
+      list2,
+      list2Width,
+      list2Height,
+      list2XCoord,
+      list2YCoord,
+    );
+
+    // Construct pivot cell.
+    final double pivotXCoord =
+        isHorizontalRectangle ? x + list1Width : x + list2Width;
+    final double pivotYCorrd = isHorizontalRectangle ? y : y + list1Height;
+    final Cell pivotCell = Cell(
+      key: UniqueKey(),
+      width: pivotBestWidth,
+      height: pivotBestHeight,
+      x: pivotXCoord,
+      y: pivotYCorrd,
+      node: pivotDataReference,
+      onTap: () => cellOnTap(pivotDataReference.name),
+      levelsVisible: widget.levelsVisible - 1,
+    );
+
+    // Construct list 3 sub-treemap.
+    final double list3Ratio = list3Size / totalSize;
+    final double list3Width =
+        isHorizontalRectangle ? list3Ratio * width : width;
+    final double list3Height =
+        isHorizontalRectangle ? height : list3Ratio * height;
+    final double list3XCoord =
+        isHorizontalRectangle ? x + list1Width + pivotBestWidth : x;
+    final double list3YCoord =
+        isHorizontalRectangle ? y : y + list1Height + pivotBestHeight;
+    List<Cell> list3Cells;
+    list3Cells = buildTreemap(
+      list3,
+      list3Width,
+      list3Height,
+      list3XCoord,
+      list3YCoord,
+    );
+
+    return list1Cells + [pivotCell] + list2Cells + list3Cells;
+  }
+
+  String nodeSizeText() {
+    final int size = rootNode.byteSize;
+    final double sizeInKB = size / 1024;
```

Comment:
look at using printKb and printMb methods from utils.dart instead of doing all this manually

Reply:
Done